src/Form/index.tsx

```diff
@@ -879,7 +878,7 @@ function Form({
     if (typeof onSubmit === 'function') {
       let stepChanged = false;
       await runUserCallback(onSubmit, () => ({
-        submitFields: formattedFields,
+        submitFields: getNonHiddenFields(activeStep, formattedFields),
```

Comment:
can this logic just live in the formatStepFields function?

Reply:
formatAllFormFields calls formatStepFields per step, so that would remove the conditionally hidden fields from more things than we want

Reply:
I guess we can just add a prop to formatStepFields